packages/styled-components/src/sheet/types.js

```diff
@@ -16,12 +16,16 @@ export interface GroupedTag {
   insertRules(group: number, rules: string[]): void;
   clearGroup(group: number): void;
   getGroup(group: number): string;
-  length: number;
+  groups: number;
 }
 
-export interface HoistedTag extends GroupedTag {
-  hoistedTag: GroupedTag;
-  normalTag: GroupedTag;
+/** Group-aware Tag that hoists @import rules utilizing GroupedWindowedTags */
+export interface HoistedTag {
```

Comment:
Not sure why the extends was removed here?

Reply:
I removed the extends mainly because I couldn't think of a way to implement the `getGroup` function that `GroupedTag` has in a way that keeps hoisting possible (Note that this is a PR into your PR), so I removed that function and replaced it with one that can split out the hoisted rules from the normal rules in the return value. I'm not aware enough of how to do things in flow to extend but only implement part of the interface (if that's even reasonable/possible) :)

Reply:
Ah alright :+1: cool, let's leave it then for now. I suppose the API will change anyway once I find time to refactor this